File: src/TGDevkit-oc/TGDevKitPro-oc/common/TGDevKit-oc/base/src/TGFileUtil/Class/TGFileUtil.m
//
//  TGFileUtil.m
//  TGFileUtil
//
//  Created by toad on 2021/08/30.
//

#import "TGFileUtil.h"

bool isNullValue(NSObject* obj,Class aClass){
    if (NULL != obj && [obj isKindOfClass:aClass]) {
        return true;
    }
    return false;
}

@implementation TGFileUtil

+ (NSString *) pathWith:(NSString *)filePath basePath:(NSString *)basePath{
    
    if (!isNullValue(filePath,[NSString class])){
        return NULL;
    }
    if ([filePath isAbsolutePath]) {
        return filePath;
    }
    if (!isNullValue(basePath,[NSString class])){
        return filePath;
    }
    return [basePath stringByAppendingPathComponent:filePath];
}

+ (NSString *) documentPathWithFilePath:(NSString *)aPath{
    return [self pathWith:aPath basePath:[NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES) firstObject]];
}

+ (NSString *) resourcePath:(NSString *)filePath bundle:(NSBundle *)bundle{
    if (isNullValue(bundle, [NSBundle class])) {
        return [self pathWith:filePath basePath:[bundle resourcePath]];
    }
    else {
        return [self pathWith:filePath basePath:[[NSBundle mainBundle] resourcePath]];
    }
}

+ (NSString *) resourcePath:(NSString *)filePath{
    return [self resourcePath:filePath bundle:NULL];
}
@end

File: src/TGDevKitPro-swift/TGDevKitPro-swift/common/TGDevKit/base/file/TGFileUtil.swift
//
//  TGFileUtil.swift
//  TG Develop Tools iOS
//
//  Created by toad on 2023/3/22.
//

import Foundation


class TGFileUtil: NSObject {
    
//    func isNullValue(obj: NSObject, aClass: AnyClass) -> Bool {
//        if(0 == obj.accessibilityElementCount()){
//            return true
//        }
//        return false;
//    }
    
    func filePath(filePath : String, basePath basePathOrNil: String?) -> String {
//        self.isKind(of: NSObject.self)
        return ""
    }
    
    func documentPath(aPath: String) -> String {
        let document = NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true).first
        return self.filePath(filePath: aPath, basePath: document)
    }
    
    func resourcePath(filePath: String,  bundle nibBundleOrNil: Bundle?) -> String {
        return ""
    }
    func resourcePath(filePath: String) -> String{
        return self.resourcePath(filePath: filePath, bundle: nil)
    }
}
